### ISSUE TITLE
Share singularity images using CVMFS

### ISSUE THREAD

**Issue by lecorguille** [nf-core/configs#184]
Hi here,

Maybe it was already suggested, somewhere?
Have you ever think about sharing the singularity images using the CernVM File System. It allow to provide files on the network based on a web of mirrors.

The Galaxy project is using this technology to share databanks, singularity images, configs accross Galaxy instances.
https://galaxyproject.org/blog/2019-02-cvmfs/

More easy to suggest than implement. So far, I'm just a client (and soon a stratum 1) and never try to build something from scratch.

My 2 cents

**Comment by MaxUlysse** [comment]
If you mean the images hosted here: https://depot.galaxyproject.org/singularity
We actually starting to do that with the DSL2 implementation that is currently ongoing for some pipelines:
- repo for modules: https://github.com/nf-core/modules
- rnaseq: https://github.com/nf-core/rnaseq/tree/dev
- chipseq: https://github.com/nf-core/chipseq/tree/dev
- sarek: https://github.com/nf-core/sarek/tree/dsl2

**Comment by lecorguille** [comment]
Great news!

**Comment by MaxUlysse** [comment]
I think the talk by Björn Grüning we had at our last hackathon (https://www.youtube.com/watch?v=_tkd_hnVYUk), convinced us to make the switch.

**Comment by apeltzer** [comment]
Its a bit of a different point - Björn talked about it last year on a different meeting with @sven1103 and me, too. The point of this CernVM File System is that you simply have this mounted at your local institution and can then use the images "as if they were there" - being downloaded locally only if they are used. So its more like a global cache than what we even do with DSLv2 - might be something we should keep in mind, as it might be very interesting for some people to learn about it: https://kt.cern/technologies/cern-vm-fs 

tl;dr: We are using biocontainers, but not shipping them via CVMFS - it might be however very interesting to have that supported at some point.

**Comment by MaxUlysse** [comment]
Oh, sorry about that, I misunderstood the question then.
Could be good to move that issue in tools or modules then?

**Comment by apeltzer** [comment]
No you were entirely right about the move to Biocontainers in DSlv2 which aims in the exact same direction - CVMFS would just make sharing these containers a bit more easy, as they would just magically appear on a network mount in your HPC ❤️  

Yes, maybe we should move this to modules or tools - not sure which one is more suitable?
